core/kogitoservice/app_configmap_reconciler.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2021 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kogitoservice
+
+import (
+	"github.com/RHsyseng/operator-utils/pkg/resource"
+	"github.com/kiegroup/kogito-operator/api"
+	"github.com/kiegroup/kogito-operator/core/framework"
+	"github.com/kiegroup/kogito-operator/core/framework/util"
+	"github.com/kiegroup/kogito-operator/core/infrastructure"
+	"github.com/kiegroup/kogito-operator/core/manager"
+	"github.com/kiegroup/kogito-operator/core/operator"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"reflect"
+	"time"
+)
+
+// AppConfigMapReconciler ...
+type AppConfigMapReconciler interface {
+	Reconcile() (time.Duration, error)
+}
+
+type appConfigMapReconciler struct {
+	operator.Context
+	instance            api.KogitoService
+	configMapHandler    infrastructure.ConfigMapHandler
+	deltaProcessor      DeltaProcessor
+	infraManager        manager.KogitoInfraManager
+	appConfigMapHandler AppConfigMapHandler
+}
+
+// NewAppConfigMapReconciler ...
+func NewAppConfigMapReconciler(context operator.Context, instance api.KogitoService, infraHandler manager.KogitoInfraHandler) AppConfigMapReconciler {
+	return &appConfigMapReconciler{
+		Context:             context,
+		instance:            instance,
+		configMapHandler:    infrastructure.NewConfigMapHandler(context),
+		deltaProcessor:      NewDeltaProcessor(context),
+		infraManager:        manager.NewKogitoInfraManager(context, infraHandler),
+		appConfigMapHandler: NewAppConfigMapHandler(context),
+	}
+}
+
+func (c *appConfigMapReconciler) Reconcile() (reconcileInterval time.Duration, err error) {
```

Comment:
not sure if crateing a new reconcile for every config map kind is a good idea.
is it really necessary? can't we have just one reconciler to all config maps managed by the operator?

Reply:
These are only internal reconciler which work as a service. Kogitoruntimr_controller receive reconcile request from olm and handed over to these service to perform its task.